doc/source/reference/routines.linalg.rst

```diff
@@ -30,10 +30,26 @@ flexible broadcasting options.  For example, `numpy.linalg.solve` can handle
 "stacked" arrays, while `scipy.linalg.solve` accepts only a single square
 array as its first argument.
 
+.. note::
+
+   The term *matrix* as it is used on this page indicates a 2d `numpy.array`
+   object, and *not* a `numpy.matrix` object. The latter is no longer
+   recommended, even for linear algebra. See
+   :ref:`the matrix object documentation<matrix-objects>` for
+   more information.
+
+The ``@`` operator
+------------------
+
+Introduced in NumPy 1.10.0, the ``@`` and ``@=`` operators are preferable to
+other methods when computing the matrix product between 2d arrays. The
+:func:`numpy.matmul` function implements the semantics of the ``@`` operator.
```

Comment:
```suggestion
:func:`numpy.matmul` function implements the ``@`` operator.
```
I guess there are two things that could be said here:
* `matmul` implements the semantics of `@` prescribed by  PEP465 (assuming that does prescribe semantics)
* `a @ b` is equivalent to `matmul(a, b)`

Since this doesn't mention the PEP, the latter sounds more likely, hence my suggestion.

Reply:
+1 for not mentioning the PEP. PEPs are proposals, not meant as documentation, and on average pretty inscrutable.